### PR TITLE
Add log search API server

### PR DESCRIPTION
**Pull request by donatello**
Initial change to add log search API server program.

### CODE REVIEW

**Review by krisis — COMMENTED** [review]
Overall changes look good to me. There are some debug print statements that need to be removed.

File: logsearchapi/server/db.go [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+		timeRangeOp = ">="
+		timeOrder = "ASC"
+	}
+
+	jw := json.NewEncoder(w)
+	switch s.Query {
+	case rawQ:
+		timeRangeClause := fmt.Sprintf("event_time %s '%s'", timeRangeOp, s.TimeStart.Format(time.RFC3339Nano))
+		q := logEventSelect.build(auditLogEventsTable.Name, timeRangeClause, timeOrder)
+		fmt.Println(q)
```
Comment (donatello) [inline logsearchapi/server/db.go:325]:
```suggestion
```

File: logsearchapi/server/server.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+//
+//   POST /api/ingest?token=xxx
+//
+// The json body represents the Audit log data. If it is an empty object the
+// request is ignored but returns success.
+func (ls *LogSearch) ingestHandler(w http.ResponseWriter, r *http.Request) {
+	// Request is assumed to be authenticated at this point.
+
+	// FIXME(aditya): Remove this printing used for debugging.
+	err := printReq(r)
```
Comment (donatello) [inline logsearchapi/server/server.go:91]:
```suggestion
	// err := printReq(r)
```

**Comment by donatello** [comment]
This is ready for review - removed more useless print statements.

**Review by krisis — APPROVED** [review]

File: logsearchapi/server/server.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+	err = ls.DBClient.InitDBTables(context.Background())
+	if err != nil {
+		return nil, fmt.Errorf("Error initializing tables: %v", err)
+	}
+
+	// Initialize muxer
+	ls.ServeMux = http.NewServeMux()
+	ls.HandleFunc("/status", func(w http.ResponseWriter, r *http.Request) {})
+	ls.HandleFunc("/api/ingest", authorize(ls.ingestHandler, ls.AuditAuthToken))
+	ls.HandleFunc("/api/query", authorize(ls.queryHandler, ls.QueryAuthToken))
```
Comment (nitisht) [inline logsearchapi/server/server.go:71]: Should we version the API endpoint? e.g. `/api/v1/query`

**Review by nitisht — APPROVED** [review]
Minor nit, otherwise LGTM. 

Are there plans to integrate this with the operator natively, e.g. a flag in Operator yaml like `requestLogSearch` to enable whole setup (PG + LogSeach + MinIO Configured to send events to LogSearch)?

File: logsearchapi/server/server.go (hunk 3 above)
Comment (donatello) [reply logsearchapi/server/server.go:71]: Yes, we will do this subsequently. Just want to get this in, to make further changes small and easy to review.

**Comment by donatello** [comment]
> Minor nit, otherwise LGTM.
> 
> Are there plans to integrate this with the operator natively, e.g. a flag in Operator yaml like `requestLogSearch` to enable whole setup (PG + LogSeach + MinIO Configured to send events to LogSearch)?

Yes #339 requires this PR and sets up the operator side integration.

**Pull request merged**